tools/DevTools/CommonRepo.psm1

```diff
@@ -0,0 +1,96 @@
+﻿<#
+ .Synopsis
+  Connects azure-powershell repo to azure-powershell-common repo for debugging.
+
+ .Description
+  Connects azure-powershell repo to azure-powershell-common repo for debugging.
+
+ .Parameter CommonRepoPath
+  Path to the common repo. Relative or absolute.
+
+ .Example
+   Connect-CommonRepo -CommonRepoPath ..\azure-powershell-common
+#>
+function Connect-CommonRepo {
+  [CmdletBinding()]
+  param(
+    [Parameter(Mandatory)]
+    [system.string]
+    ${CommonRepoPath}
+  )
+
+  Write-Host "1/2 Adding common projects to sln and csproj"
+
+  $csprojs = @(
+    '.\src\Authentication.Abstractions\Authentication.Abstractions.csproj',
+    '.\src\Common\Common.csproj',
+    '.\src\ResourceManager\ResourceManager.csproj',
+    '.\src\Graph.Rbac\Graph.Rbac.csproj'
+  )
+  $CommonRepoPath = (Resolve-Path $CommonRepoPath).Path
+  $RepoRoot = "$PSScriptRoot/../.."
+  Push-Location "$RepoRoot/src/Accounts"
+  try {
+    foreach ($csproj in $csprojs) {
+      $csproj = [System.IO.Path]::Combine($CommonRepoPath, $csproj)
+      $csproj = [System.IO.Path]::GetFullPath($csproj)
+      dotnet sln add $csproj
+      if ($LASTEXITCODE -ne 0) {
+        throw "Failed to add $csproj to Accounts.sln"
+      }
+      dotnet add ./Accounts/Accounts.csproj reference $csproj
+      if ($LASTEXITCODE -ne 0) {
+        throw "Failed to add $csproj to Accounts.csproj"
+      }
+      dotnet add ./Authentication/Authentication.csproj reference $csproj
+      if ($LASTEXITCODE -ne 0) {
+        throw "Failed to add $csproj to Authentication.csproj"
+      }
+
+    }
+  }
+  finally {
+    Pop-Location
+  }
+
+
+  Write-Host "2/2: Remove the dependency of those common projects from .targets file"
+
+  $Patterns = @(
+    '<PackageReference Include="Microsoft.Azure.PowerShell.Authentication.Abstractions"',
+    '<PackageReference Include="Microsoft.Azure.PowerShell.Common"',
+    '<PackageReference Include="Microsoft.Azure.PowerShell.Clients.Graph.Rbac"',
+    '<PackageReference Include="Microsoft.Azure.PowerShell.Clients.ResourceManager"',
+    '<PackageReference Include="Microsoft.Azure.PowerShell.Common.Share"'
```

Comment:
These 5 projects might not be enough in some cases, for example when debugging with Az.Compute.
I should consider adding all common projects in the script.

Reply:
Now all the common projects are added.